provider/maas/environ.go

```diff
@@ -2171,7 +2171,23 @@ func (env *maasEnviron) allocateContainerAddresses2(hostInstanceID instance.Id,
 			subnetCIDRToSubnet[subnet.CIDR()] = subnet
 		}
 	}
+	// map from the source subnet (what subnet is the device in), to what
+	// static routes should be used.
+	subnetToStaticRoutes := make(map[string][]gomaasapi.StaticRoute)
+	staticRoutes, err := env.maasController.StaticRoutes()
+	if err != nil {
+		// Warning?
```

Comment:
So my original thought was that I wasn't sure what versions of MAAS supported static-routes as an API, though the docs do seems to say that 2.0 onward support it.
That said, if you aren't using static routes, all the other work is still valid, so an error in accessing them didn't seem like it should be fatal vs anything else creating a container. But errors here should be rare, so I'm fine just propagating them up.  I switched this to return errors.Annotate() which handles the next item as well.